toolchest_client/tools/tool.py

```diff
@@ -189,14 +189,15 @@ def _validate_args(self):
 
         if self.inputs is None:
             raise ValueError("No input provided.")
-        if self.output_path and not os.access(
+        if self.output_path and not path_is_s3_uri(self.output_path) and not os.access(
```

Comment:
I've seen `if self.output_path and not path_is_s3_uri(self.output_path)` five times in this PR – might be worth specifying an instance attribute of `has_local_output` or similar?

Reply:
We've already removed `is_s3` as an input file attribute which seems very similar to `has_local_output` I could extract it to a function so it's only fully expanded in 1 spot. Also allows for us to change that condition compared to an attribute that's more locked in.

Reply:
We've already removed `is_s3` as an attribute of input file which seems very similar to `has_local_output`. I can extract it to a function instead of an attribute so it's only fully expanded in one spot instead. A function is also more flexible if we want to update the condition in the future.

Reply:
Sounds good – I trust your judgement on either, I also think it would be okay to leave as-is